scipy/spatial/transform/_rotation.py

```diff
@@ -2035,8 +2035,10 @@ def mean(self, weights: ArrayLike | None = None) -> Rotation:
         >>> r.mean().as_euler('zyx', degrees=True)
         array([0.24945696, 0.25054542, 0.24945696])
         """
-        mean = self._backend.mean(self._quat, weights=weights)
-        return Rotation._from_raw_quat(mean, xp=self._xp, backend=self._backend)
+        cython_compatible = self._quat.ndim < 3
+        backend = select_backend(self._xp, cython_compatible=cython_compatible)
+        mean = backend.mean(self._quat, weights=weights)
+        return Rotation._from_raw_quat(mean, xp=self._xp, backend=backend)
```

Comment:
We don't need to re-determine the backend again, `self._backend` already accounts for `self._quat`'s dimension. We only need to double-check if we interact with new inputs of unknown dimensions (i.e. when multiplying with another quat).